File: Algorithms/C++/0350/0350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        multiset<int> s1, s2;
        for(int x : nums1) {
            s1.insert(x);
        }
        for(int x : nums2) {
            s2.insert(x);
        }
        vector<int> res;
        set_intersection(s1.begin(), s1.end(), s2.begin(), s2.end(), back_inserter(res));
        return res;
    }
};

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        map<int, int> map1, map2;
        for (int num : nums1) {
            ++map1[num];
        }
        for (int num: nums2) {
            ++map2[num];
        }
        vector<int> v;
        for (auto p : map1) {
            for (int i = 0; i < min(p.second, map2[p.first]); i++) {
                v.push_back(p.first);
            }
        }
        return v;
    }
};

File: Algorithms/Python/0350/0350.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        dict1 = {}
        dict2 = {}
        for num in nums1:
            if num in dict1:
                dict1[num] += 1
            else:
                dict1[num] = 1
        for num in nums2:
            if num in dict2:
                dict2[num] += 1
            else:
                dict2[num] = 1
        res = []
        for key, value in dict1.items():
            if key in dict2:
                for i in range(min(value, dict2[key])):
                    res.append(key)
        return res
